Source/control.cpp

```diff
@@ -1719,9 +1722,35 @@ static int DrawDurIcon4Item(CelOutputBuffer out, ItemStruct *pItem, int x, int c
 			break;
 		}
 	}
-	if (pItem->_iDurability > 2)
-		c += 8;
-	CelDrawTo(out, x, -17 + PANEL_Y, pDurIcons, c, 32);
+
+	int y = -17 + PANEL_Y;
+
+	//Calculate how much of the icon should be gold and red
+	int height = 32;
+	int amount;
+	if (pItem->_iDurability <= DURABILITY_RED)
+		amount = 0;
+	else {
+		if (!sgOptions.Graphics.bDurabilityIconGradual)
+			amount = height;
+		else {
+			int cur = pItem->_iDurability - DURABILITY_RED;
+			amount = (height * cur) / (DURABILITY_GOLD - DURABILITY_RED);
+		}
+	}
+
+	//Draw icon
+	if (amount) {
+		CelOutputBuffer buf = GlobalBackBuffer();

```

Comment:
Avoid abbreviations.

Reply:
I think it's called `out` almost everywhere else, technically not an abbreviation, but :shrug: This will all go away in a couple of refactors of the drawing code.